### PR TITLE
replace sensor by input

### PR DESCRIPTION
**Pull request by JeremyActronika**
- Delete sensor interface in libtup
- Create libtup interface for input

### CODE REVIEW

File: include/libtup.h [hunk 1]
```diff
@@ -110,13 +110,13 @@ typedef struct
...
- * Get/Set sensor value structure
+ * Get/Set input value structure
  */
 typedef struct
 {
-    uint8_t sensor_id;        /**< sensor id */
-    uint16_t sensor_value;    /**< sensor value */
-} TupSensorValueArgs;
+    uint8_t input_id;        /**< input id */
+    uint16_t input_value;    /**< input value */
```
Comment (thactronika) [inline include/libtup.h]: you shall declare it as a int32_t to correspond to libart

**Comment by azanelli-actronika** [comment]
Related to ActronikaSAS/libart#48

**Comment by azanelli-actronika** [comment]
We still have some pending comment

File: include/libtup.h (hunk 1 above)
Comment (JeremyActronika) [reply include/libtup.h]: Done

File: tools/tupctl.c [hunk 2]
```diff
@@ -429,6 +439,99 @@ static int do_set_sensor_value(int argc, char *argv[])
...
+    if (ret != 1) {
+        printf("failed to parse arguments\n");
+        return -EINVAL;
+    }
+    switch(state_activation) {
+        case 0:
+            printf("Internal Sensors deactivated\n");
+            break;
+        case 1:
+            printf("Internal Sensors activated\n");
```
Comment (azanelli-actronika) [inline tools/tupctl.c]: these prints should not be in the preterit mood, but rather in the imperative, ie:
`activate internal sensors`

as their are not activated yet.

**Comment by azanelli-actronika** [comment]
Could we add a commit to update examples ?

File: docs/messages.txt [hunk 3]
```diff
@@ -12,8 +12,10 @@ Message list and ids:
     14 - GET_PARAMETER
     15 - SET_PARAMETER
     16 - BIND_EFFECT
-    17 - GET_SENSOR_VALUE
-    18 - SET_SENSOR_VALUE
+    17 - GET_SENSOR_VALUE @DEPRECATED
+    18 - SET_SENSOR_VALUE @DEPRECATED
```
Comment (JeremyActronika) [inline docs/messages.txt]: Add documentation about activate sensor

**Comment by thactronika** [comment]
Examples updated, doc updated on activation for internal sensors, still missing the one on inputs

File: docs/messages.txt [hunk 4]
```diff
@@ -147,36 +148,51 @@ BIND_EFFECT:
     expected response:
         ACK, ERROR
 
-GET_SENSOR_VALUE:
+GET_INPUT_VALUE:
```
Comment (azanelli-actronika) [inline docs/messages.txt]: we should just add section as GET_SENSOR_VALUE still exist

File: docs/messages.txt [hunk 5]
```diff
@@ -147,36 +148,51 @@ BIND_EFFECT:
...
     arguments:
-        n * uint8_t - sensor from which value should be get
+        uint8_t - slot of the effect
+        n * uint8_t - input value from which effect should be get
 
     expected response:
         RESP_SENSOR, ERROR
 
-SET_SENSOR_VALUE:
+SET_INPUT_VALUE:
```
Comment (azanelli-actronika) [inline docs/messages.txt]: We should just add section as SET_SENSOR_VALUE still exist

**Pull request merged**